compio/tests/io.rs

```diff
@@ -159,3 +165,179 @@ async fn writev_at() {
     assert_eq!(len, 3);
     assert_eq!(dst, [0, 0, 1, 1, 4]);
 }
+
+struct RepeatOne(u8);
+
+impl AsyncRead for RepeatOne {
+    async fn read<B: IoBufMut>(&mut self, mut buf: B) -> compio_buf::BufResult<usize, B> {
+        let slice = buf.as_mut_slice();
+        if !slice.is_empty() {
```

Comment:
This test is quite confusing. Can you elaborate more on thie?

Reply:
To add test for *_exact, I need to ensure that `read` doesn't read all data once into the buffer.

Reply:
But why a zero at beginning?

Reply:
Just to make sure that `read_at` works well